tests/test_livysession.py

```diff
@@ -436,6 +437,72 @@ def test_create_sql_hive_context_happens_once(self):
                in http_client.post_statement.call_args_list
         assert len(http_client.post_statement.call_args_list) == 1
 
+    def test_create_sql_context_failure(self):
+        kind = constants.SESSION_KIND_SPARK
+        http_client = MagicMock()
+        ipython_display = MagicMock()
+
+        http_client.post_session.return_value = self.session_create_json
+        self.post_statement_responses = [self.post_statement_json,
+                                         self.post_statement_json]
+        http_client.post_statement.side_effect = self._next_statement_response_post
+        http_client.get_session.return_value = self.ready_sessions_json
+        self.get_statement_responses = [self.running_statement_json,
+                                        self.ready_statement_json,
+                                        self.ready_statement_json]
+        http_client.get_statement.side_effect = self._next_statement_response_get
+        conf.override_all({
+            "status_sleep_seconds": 0.01,
+            "statement_sleep_seconds": 0.01
+        })
+        session = self._create_session(kind=kind, http_client=http_client)
+        session.ipython_display = ipython_display
+        conf.load()
+
+        session._get_sql_context_creation_command = MagicMock()
+
+        session._get_sql_context_creation_command.return_value.execute.return_value = (True, "Success")
+        session.created_sql_context = None
+        session.create_sql_context()
+        assert session.created_sql_context
+
+        session._get_sql_context_creation_command.return_value.execute.return_value = (False, "Exception")
+        session.created_sql_context = None
+
+        try:
+            session.create_sql_context()
+            assert False
+        except ValueError as ex:
+            assert str(ex) == "Failed to create the SqlContext.\nError, '{}'".format("Exception")
+            assert session.created_sql_context is None
+
+        kernelMagics = KernelMagics(shell=None)
```

Comment:
Why is this here and not in `test_kernel_magics` (the file where we're doing the rest of the testing of kernel magics)?